api/src/main/java/javax/enterprise/inject/se/SeContainerInitializer.java

```diff
@@ -0,0 +1,317 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package javax.enterprise.inject.se;
+
+import javax.enterprise.inject.spi.AnnotatedType;
+import javax.enterprise.inject.spi.Bean;
+import javax.enterprise.inject.spi.Extension;
+import java.lang.annotation.Annotation;
+import java.util.Map;
+import java.util.ServiceConfigurationError;
+import java.util.ServiceLoader;
+
+/**
+ * Container initializer to bootstrap CDI in Java SE.
+ * It is obtained by calling the {@link SeContainerInitializer#getInstance()} static method
+ * <p>
+ * <p>
+ * Typical usage looks like this:
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainer<Object> container = SeContainerInitializer.getInstance().initialize();
+ * container.select(Foo.class).get();
+ * container.event().select(Bar.class).fire(new Bar());
+ * container.close();
+ * </pre>
+ * <p>
+ * <p>
+ * The {@link SeContainer} interface implements AutoCloseable:
+ * </p>
+ * <p>
+ * <pre>
+ * try (SeContainer<Object> container = SeContainerInitializer.getInstance().initialize()) {
+ *     container.select(Foo.class).get();
+ * }
+ * </pre>
+ * <p>
+ * <p>
+ * By default, the discovery is enabled so that all beans from all discovered bean archives are considered. However, it's possible to define a "synthetic" bean
+ * archive, or the set of bean classes and enablement respectively:
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainer<Object> container = SeContainerInitializer.getInstance().beanClasses(Foo.class, Bar.class).alternatives(Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * Moreover, it's also possible to disable the discovery completely so that only the "synthetic" bean archive is considered:
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainer<Object> container = SeContainerInitializer.getInstance().disableDiscovery().beanClasses(Foo.class, Bar.class).initialize());
+ * </pre>
+ * <p>
+ * <p>
+ * <p>
+ * In the same manner, it is possible to explicitly declare interceptors, decorators, extensions and implementation specific options using the builder.
+ * </p>
+ * <p>
+ * <pre>
+ * SeContainerInitializer containerInitializer = SeContainerInitializer.getInstance()
+ *    .disableDiscovery()
+ *    .addPackages(Main.class, Utils.class)
+ *    .addInterceptors(TransactionalInterceptor.class)
+ *    .addProperty("property", true);
+ * SeContainer<Object> container = container.initialize();
+ * </pre>
+ *
+ * @author Antoine Sabot-Durand
+ * @author Martin Kouba
+ * @author John D. Ament
+ * @since 2.0
+ */
+public abstract class SeContainerInitializer {
+
+    protected static volatile SeContainerInitializer seContainerInitializer = null;
+
+
+    /**
+     * Returns an instance of {@link SeContainerInitializer}
+     * Each call returns a new instance
+     *
+     * @return a new SeContainerInitializer instance.
+     * @throws IllegalStateException if called in a Java EE container
+     */
+    public static SeContainerInitializer getInstance() {
```

Comment:
+1 on renaming to newInstance method name.

Many thanks,
Emily
===========================
Emily Jiang
WebSphere Application Server, CDI Development Lead
 
MP 211, DE3A20, Winchester, Hampshire, England, SO21 2JN
Phone:  +44 (0)1962 816278  Internal: 246278

Email: emijiang@uk.ibm.com 
Lotus Notes: Emily Jiang/UK/IBM@IBMGB




From:   Antoine Sabot-Durand <notifications@github.com>
To:     cdi-spec/cdi <cdi@noreply.github.com>, 
Cc:     Emily Jiang/UK/IBM@IBMGB, Mention <mention@noreply.github.com>
Date:   30/06/2016 08:55
Subject:        Re: [cdi-spec/cdi] CDI-568 enhancing Java SE bootstrap API 
(#290)



In 
api/src/main/java/javax/enterprise/inject/se/SeContainerInitializer.java:
> + * @author John D. Ament
> + * @since 2.0
> + */
> +public abstract class SeContainerInitializer {
> +
> +    protected static volatile SeContainerInitializer 
seContainerInitializer = null;
> +
> +
> +    /**
> +     * Returns an instance of {@link SeContainerInitializer}
> +     * Each call returns a new instance
> +     *
> +     * @return a new SeContainerInitializer instance.
> +     * @throws IllegalStateException if called in a Java EE container
> +     */
> +    public static SeContainerInitializer getInstance() {

Of course I have no objection to rename the method, if everybody find it 
clearer.
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

Unless stated otherwise above:
IBM United Kingdom Limited - Registered in England and Wales with number 
741598. 
Registered office: PO Box 41, North Harbour, Portsmouth, Hampshire PO6 3AU